packages/host/app/components/matrix/register-user.gts

```diff
@@ -698,6 +700,7 @@ export default class RegisterUser extends Component<Signature> {
     // If access_token and device_id are present, RegisterResponse matches LoginResponse
     // except for the optional well_known field
     if (auth.access_token && auth.device_id) {
+      await this.router.refresh();
```

Comment:
i'm not 100% sure that the calls that happen after this are executed, since this component is destroyed after refreshing the router. I would suggest moving this to after `this.args.setMode('login');`